db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogiteminitialization.rb

```diff
@@ -81,17 +81,24 @@ def assign_service_tag(tag_category, tag_value)
   @service.tag_assign("#{tag_category}/#{tag_value}")
 end
 
-def get_vm_name(dialog_options_hash, prov)
-  log_and_update_message(:info, "Processing get_vm_name", true)
+def set_vm_name(dialog_options_hash, prov)
+  log_and_update_message(:info, "Processing set_vm_name", true)
   new_vm_name = dialog_options_hash.fetch(:vm_name, nil) || dialog_options_hash.fetch(:vm_target_name, nil)
+  if new_vm_name.blank?
+    set_all_vm_name_attrs(prov, prov.get_option(:vm_target_name))
+    log_and_update_message(:info, "Using default vm name: #{prov.get_option(:vm_target_name)}", true)
+  else
+    set_all_vm_name_attrs(prov, new_vm_name)
+    log_and_update_message(:info, "Setting vm name to: #{prov.get_option(:vm_target_name)}", true)
+  end
+  log_and_update_message(:info, "Processing set_vm_name...Complete", true)
+end
```

Comment:
@gmcculloug 
When you provision multiple VM's the names are setup first, then the catalog item initialization code would end up overwriting the name. The logic in catalog item initialization works only for single VM because it updated the dialog_options_hash instead of setting the vm_name in the provision object.